src/main/java/jenkins/plugins/slack/webhook/WebhookEndpoint.java

```diff
@@ -0,0 +1,245 @@
+package jenkins.plugins.slack.webhook;
+
+
+import jenkins.model.Jenkins;
+import jenkins.model.GlobalConfiguration;
+
+import hudson.Extension;
+
+import hudson.model.Build;
+import hudson.model.Result;
+import hudson.model.Project;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.UnprotectedRootAction;
+
+import hudson.security.ACL;
+
+import javax.servlet.ServletException;
+
+import java.io.IOException;
+
+import java.util.List;
+import java.util.ArrayList;
+
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import jenkins.plugins.slack.webhook.model.JsonResponse;
+import jenkins.plugins.slack.webhook.model.SlackPostData;
+import jenkins.plugins.slack.webhook.model.SlackTextMessage;
+import jenkins.plugins.slack.webhook.model.SlackWebhookCause;
+
+import jenkins.plugins.slack.webhook.exception.CommandRouterException;
+import jenkins.plugins.slack.webhook.exception.RouteNotFoundException;
+
+
+
+
+@Extension
+public class WebhookEndpoint implements UnprotectedRootAction {
+
+    private String slackUser;
+
+    private GlobalConfig globalConfig;
+
+    private static final Logger LOGGER =
+        Logger.getLogger(WebhookEndpoint.class.getName());
+
+    public WebhookEndpoint() {
+        globalConfig = GlobalConfiguration.all().get(GlobalConfig.class);
+    }
+
+    @Override
```

Comment:
The Jenkins project seems to want Java 1.7 and later.  ref https://jenkins-ci.org/blog/2015/04/06/good-bye-java6/

Here's a link to another plugin I support updating the compatibility.

https://github.com/jenkinsci/github-oauth-plugin/pull/42